src/plugins/graph/adornments/plotted-function/plotted-variables/plotted-variables-adornment-model.ts

```diff
@@ -0,0 +1,108 @@
+import { destroy, getSnapshot, IAnyStateTreeNode, Instance, types } from "mobx-state-tree";
+import { Variable, VariableType } from "@concord-consortium/diagram-view";
+
+import { getTileModel } from "../../../../../models/document/shared-model-document-manager";
+import { getSharedModelManager } from "../../../../../models/tiles/tile-environment";
+import { uniqueId } from "../../../../../utilities/js-utils";
+import { SharedVariables, SharedVariablesType } from "../../../../shared-variables/shared-variables";
+import { IAdornmentModel } from "../../adornment-models";
+import { kPlottedVariablesType } from "./plotted-variables-adornment-types";
+import { PlottedFunctionAdornmentModel } from "../plotted-function-adornment-model";
+
+function getSharedVariables(node: IAnyStateTreeNode) {
+  const sharedModelManager = getSharedModelManager(node);
+  if (sharedModelManager?.isReady) {
+    const tile = getTileModel(node);
+    if (tile) {
+      const sharedVariables = sharedModelManager.findFirstSharedModelByType(SharedVariables, tile.id);
+      if (sharedVariables) return sharedVariables as SharedVariablesType;
+    }
+  }
+}
+
+export const PlottedVariablesInstance = types.model("PlottedVariablesInstance", {})
```

Comment:
Based on Kirk's comments, maybe we should not call this an "Instance" since that terminology is used for something else in CODAP.